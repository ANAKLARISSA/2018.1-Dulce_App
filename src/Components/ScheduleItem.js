import React from 'react';
import {Text, TouchableHighlight, View} from 'react-native';
import {scheduleItem as styles} from './styles' ;
import axios from 'axios';
import store from '../Reducers/store';
import {purple_dulce} from '../styles';

class ScheduleItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      timezone: new Date().getTimezoneOffset() / 60,
      start_time: this.props.item.start_time ? new Date(this.props.item.start_time) : '',
      end_time: this.props.item.end_time ? new Date(this.props.item.end_time) : '',
      start_time_string: '',
      end_time_string: '',
      userId: '',
      employee: '',
      specialty: 'teste',
      loading: true
    };
  }

  async componentDidMount() {
    await this.setTimeString();
    await this.setState({loading: false});
  }
  async axiosProfile() {
    const url = 'http://18.231.9.190:8083/api/profile/view/?profile_id=' + this.props.item.profile_id;
    axios.get(url,{
      headers: {
        'Authorization': 'Bearer ' + store.getState().currentUser.token
      }
    })
    .then((response) => {
      this.setState({userId: response.data.user_id, specialty: response.data.speciality});
      console.log('setor:' + response.data.sector_id);
      this.axiosUser();
    });
  }

  axiosUser() {
    const url = 'http://52.67.4.137:8083/api/user/listById?id=' + this.state.userId;
    axios.get(url,{
      headers: {
        'Authorization': 'Bearer ' + store.getState().currentUser.token
      }
    })
    .then((response) => {
      this.setState({employee: response.data.firstName + ' ' + response.data.lastName});
    });
  }

  async setTimeString() {
    if ((this.state.start_time !== '') && (this.state.end_time !== '')) {
      var st_string = (this.state.start_time.getHours() + this.state.timezone).toString().padStart(2, 0) +
       ':' + this.state.start_time.getMinutes().toString().padStart(2, 0);
      var et_string = (this.state.end_time.getHours() + this.state.timezone).toString().padStart(2, 0) +
      ':' + this.state.end_time.getMinutes().toString().padStart(2, 0);
      if ((this.state.start_time.getDate() < this.state.end_time.getDate()) ||
      (this.state.start_time.getMonth() < this.state.end_time.getMonth())) {
        et_string += '(' + this.state.end_time.getDay().toString().padStart(2, 0) + '/' +
        (this.state.end_time.getMonth() + 1).toString().padStart(2, 0) + '/' +
        (this.state.end_time.getYear()).toString().padStart(2, 0) + ')';
      }
      this.setState({start_time_string: st_string, end_time_string: et_string});
    }
    await this.axiosProfile();
  }

  renderSectorItem() {
    return (
      <TouchableHighlight
        underlayColor={purple_dulce}
        onPress={() => {this.props.onPress();}}
        style={styles.item}>
        <View style={{padding: 5}}>
        {
          this.props.isSector ? (
            <View>
              <Text>{this.state.employee}</Text>
              <Text>{this.state.start_time_string} - {this.state.end_time_string}</Text>
              <Text>{this.state.specialty}</Text>
            </View>
          ) : (
            <View>
              <Text>Hospital</Text>
              <Text>{this.state.start_time_string} - {this.state.end_time_string}</Text>
            </View>
          )
        }
        </View>
      </TouchableHighlight>
    );
  }

  render() {
    return (
      <View>{
        this.state.loading ? (<Text>Loading...</Text>)
        : this.renderSectorItem()
      }
      </View>
    );
  }
}

export default ScheduleItem;
