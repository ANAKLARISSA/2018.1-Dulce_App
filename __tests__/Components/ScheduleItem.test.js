import React from 'react';
import ScheduleItem from '../../src/Components/ScheduleItem';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
require('bezier');

Enzyme.configure({adapter: new Adapter()});

const item = {
  employee: 'Example',
  specialty: 'Passing tests',
  start_time: '09:00',
  end_time: '17:00',
  amount_of_hours: '8h'
};

it('renders correctly', () => {
  const tree = renderer
  .create(<ScheduleItem item={item}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Testing onPress', () => {
  const onPressMock = jest.fn();
  const wrapper = shallow(<ScheduleItem item={item} onPress = {onPressMock}/>);
  const button = wrapper.find('TouchableHighlight').at(0);
  button.simulate('press');
  expect(onPressMock).toBeCalled();
});
