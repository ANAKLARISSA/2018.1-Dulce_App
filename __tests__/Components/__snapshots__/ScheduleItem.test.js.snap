// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "flex": 1,
      "marginRight": 10,
      "marginTop": 17,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Example
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Passing tests
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      09:00
       - 
      17:00
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      8h
    </Text>
  </View>
</View>
`;
